plugins/woocommerce/client/admin/client/settings-payments/settings-payments-main.tsx

```diff
@@ -17,64 +21,129 @@ import { WooPaymentsGatewayData } from '~/settings-payments/types';
 import { parseScriptTag } from '~/settings-payments/utils';
 
 export const SettingsPaymentsMain = () => {
-	const [ registeredPaymentGateways, setRegisteredPaymentGateways ] =
-		useState< PaymentGateway[] >( [] );
-	const [ preferredPluginSuggestions, setPreferredPluginSuggestions ] =
-		useState< Plugin[] >( [] );
-	const [ otherPluginSuggestions, setOtherPluginSuggestions ] = useState<
-		Plugin[]
-	>( [] );
 	const [ wooPaymentsGatewayData, setWooPaymentsGatewayData ] = useState<
 		WooPaymentsGatewayData | undefined
 	>( undefined );
 	const [ installingPlugin, setInstallingPlugin ] = useState< string | null >(
 		null
 	);
+	const [ isInstalled, setIsInstalled ] = useState< boolean >( false );
+	const [ isEnabled, setIsEnabled ] = useState< boolean >( false );
 	const { installAndActivatePlugins } = useDispatch( PLUGINS_STORE_NAME );
 
 	const installedPluginSlugs = useSelect( ( select ) => {
 		return select( PLUGINS_STORE_NAME ).getInstalledPlugins();
 	}, [] );
 
+	// Make UI to refresh when plugin is installed.
+	const { invalidateResolution } = useDispatch( 'core/data' );
```

Comment:
not sure if it's the best way to do it, since we are installing plugin using another store. cc @woocommerce/ghidorah

Reply:
Yes, `invalidateResolution` is a function that is used to refresh the data in the store. However, I think you can use `invalidateResolutionForStoreSelector` from the payment suggestion store instead of using the core/data store.

And I wonder if we can call it directly when plugin is installed in `setupPlugin` function so we don't need to use `useEffect`, which would cause the component to re-render.

Reply:
Amazing suggestion! Fixed in https://github.com/woocommerce/woocommerce/pull/52824/commits/5cd0c08ff3fef6f9b0beef0a0d84009932143d89